sgx-show/src/sink/bit.rs

```diff
@@ -0,0 +1,14 @@
+use crate::data::cpuid::{CpuId, Leaf};
+
+pub struct Bit<T: Leaf> {
+    pub reg: fn(&CpuId<T>) -> u32,
+    pub bit: u8,
+}
+
+impl<T: Leaf> super::Sink for Bit<T> {
+    type Type = CpuId<T>;
+
+    fn test(&self, data: &Self::Type) -> bool {
+        (self.reg)(data) & (1 << self.bit) != 0
```

Comment:
I don't totally understand this line and `(self.reg)(data)`. `data` is a reference to a `CpuId<T>` and `reg` is a function to turn that into a `u32`? What do the parens do here?

Reply:
`self.reg` is a function pointer. So we are calling the function pointed to by `self.reg`. This is a function to extract one particular register from the CpuId struct. See, for example: https://github.com/enarx/enarx/pull/97/files#diff-187ebd96da7420d347a4eae0e18468b6R124